### ISSUE TITLE
kak-lsp stops sending requests to nimlangserver upon edit/save

### ISSUE THREAD

**Issue by jdugan6240** [kak-lsp/kak-lsp#695]
**Steps to Reproduce:**
Create a file with the following Nim code (this is what is generated with `nimble init`):

```nim
# This is just an example to get you started. A typical binary package
# uses this file as the main entry point of the application.

when isMainModule:
  echo("Hello, World!")
```

Edit the file with the language server set to [nimlangserver](https://github.com/nim-lang/langserver). Make a simple edit (say, removing the colon).

**What Happens:**
After editing, `kak-lsp` will cease to send any requests to the server until exit, thus breaking LSP functionality.d

**What Should Happen:**

`kak-lsp` should continue communicating with the server.

A log of a simple editing session is attached here: [log.txt](https://github.com/kak-lsp/kak-lsp/files/13197093/log.txt)

**Comment by krobelus** [comment]
I did `nimble install nimlangserver` using Nim `2.0.0`.
Then `nimble init` selecting defaults.
When I run `lsp-enable` I get an error popup/debug buffer with
```
kak-lsp: error from server: Server failed with Nimsuggest process crashed..
kak-lsp: error from server: Nimsuggest initialization for /home/johannes/git/kak-lsp/t/nim/src/nim.nim failed with: Nimsuggest process crashed.
```

I don't know where to find more logs, nimlangserver doesn't take options it seems

When I create a binary package with `nimble init` it doesn't build with no useful error:

```
$ nimble build --verbose
     Info:  Nimble data file "/home/johannes/.nimble/nimbledata2.json" has been loaded.
  Verifying dependencies for nim@0.1.0
   Building nim/nim using c backend
      Info: compiling nim package using /home/johannes/git/kak-lsp/t/nim/bin/nim
nimble.nim(229)          buildFromDir

    Error:  Build failed for the package: nim
     Info:  Nimble data file "/home/johannes/.nimble/nimbledata2.json" has been saved.
```

**Comment by jdugan6240** [comment]
Why is Nimble using nim 0.1.0...? That should not be happening.

At any rate, if it helps, I've run into the same thing with [lua-language-server](https://github.com/LuaLS/lua-language-server) and [ols](https://github.com/DanielGavin/ols), although that last one requires a Kakoune patch for Odin support: https://github.com/mawww/kakoune/pull/4988.

**Comment by krobelus** [comment]
yes this is a very interesting bug.. I'll try to get it working with nim or maybe lua

**Comment by krobelus** [comment]
found my problem with help from nim IRC. I had named my project "nim", apparently that shadowed and broke something

**Comment by krobelus** [comment]
I still get `Nimsuggest process crashed..` these people don't like verbose error messages huh

**Comment by krobelus** [comment]
`nimsuggest $PWD/src/totallynotnim.nim --v3 --autobind --log --debug` exits with 1, no error message.

I can't reproduce the problem with lua. Any recipe?

**Comment by jdugan6240** [comment]
OK, here's how I have Lua set up:
I am using the lua-language-server version currently in the Arch repos (3.7.0-1).
My kakrc (or at least the kak-lsp relevant bits) is set up as follows (using [kak-bundle](https://github.com/jdugan6240/kak-bundle) for plugin management):
 ```kak
bundle kak-lsp "https://github.com/kak-lsp/kak-lsp" %{
    lsp-auto-hover-enable
    lsp-inlay-hints-enable buffer
    lsp-inline-diagnostice-enable buffer
} %{
    cargo install --locked --force --path .
}
...
hook global WinSetOption filetype=go|lua|nim|odin|python %{
    lsp-enable-window
}
```
The Lua entry in my kak-lsp.toml is set up as follows (more or less copy pasted from the default kak-lsp.toml):
```toml
[language.lua]
filetypes = ["lua"]
roots = [".git", ".hg"]
command = "lua-language-server"
[language.lua.settings.Lua]
# See https://github.com/sumneko/vscode-lua/blob/master/setting/schema.json
# diagnostics.enable = true
```

With this setup, just about any Lua file causes the above issue for me.

**Comment by jdugan6240** [comment]
Just realized there are some issues with my kakrc config, lol. I'll have to fix those when I get home.

**Comment by krobelus** [comment]
Thanks.
If you can give a reproducer that works independent of your home directory, that would be splendid.
Like start with the command in https://github.com/kak-lsp/kak-lsp#default-configuration and add the minimal set of config options needed to trigger the issue

> lsp-inlay-hints-enable buffer

This doesn't really make sense unless you run it per buffer. Maybe you want `global` instead of `buffer`. I'm not even sure if there is a buffer context available at startup

**Comment by jdugan6240** [comment]
OK, when running the minimal configuration described in the README, lua-language-server and nimlsp (the Nim language server specified in the default kak-lsp.toml) worked fine. So, trying to isolate the issue, I reduced my kakrc to the following:

```kak
evaluate-commands %sh{kak-lsp --kakoune -s $kak_session}
set-option global lsp_cmd "kak-lsp -s %val{session} -vvvv --log /tmp/kak-lsp.log"
lsp-enable
```

This is, more or less, exactly what was passed to Kakoune's -e flag, and should therefore in theory offer the same result. As it turns out, it did. Then, I tried putting back everything in my kakrc piece by piece, to determine what would break it, and... it never broke. Apparently, I was encountering some kind of weird Heisenbug...

I have no idea what was wrong still, but at any rate, it seems to be working now.
